api/reporting/schemas/reporting-codegen.rb

```diff
@@ -26,64 +27,187 @@ def to_output_path(dir_name, table_name, suffix)
+             :constant => 'EmailVerificationStatus.SUBSCRIBED'
+         }
+     }
+}.freeze
+
+# strip size/kind
+def simple_mysql_type(mysql_type)
   type_pattern = Regexp.new("(?<type>\\w+)(\\(\\d+\\))?")
   match_data = mysql_type.match(type_pattern)
-  result = MYSQL_TO_BIGQUERY_TYPE[match_data[:type]]
+  result = match_data[:type]
   raise "MySQL type #{mysql_type} not recognized." if result.nil?
   result
 end
 
-excluded_fields = File.exist?(inputs[:exclude_columns]) \
-  ? File.readlines(inputs[:exclude_columns]) \
+excluded_fields = File.exist?(INPUTS[:excluded_columns]) \
+  ? File.readlines(INPUTS[:excluded_columns]) \
       .map{ |c| c.strip } \
   : []
 
 def include_field?(excluded_fields, field)
   !excluded_fields.include?(field)
 end
 
+ENTITY_MODIFIED_COLUMNS = {
+    'workspace' => {
+        'cdr_version_id' => 'cdrVersion.cdrVersionId AS cdrVersionId',
+        'creator_id' => 'creator.userId AS creatorId',
+        'needs_rp_review_prompt' => 'needsResearchPurposeReviewPrompt AS needsRpReviewPrompt'
+    }
+}
+
 ## BigQuery schema
-describe_rows = CSV.new(File.read(inputs[:describe_csv])).sort_by { |row| row[0]  }
+describe_rows = CSV.new(File.read(INPUTS[:describe_csv])).sort_by { |row| row[0]  }
+
+root_class_name = to_camel_case(table_name, true)
+TABLE_INFO = {
+    :name => table_name,
+    :instance_name => to_camel_case(table_name, false),
+    :dto_class => "BqDto#{root_class_name}",
+    :entity_class => "Db#{root_class_name}",
+    :mock => "mock#{root_class_name}",
+    :projection_interface => "Prj#{root_class_name}",
+    :sql_alias => table_name[0].downcase,
+    :enum_column_info => ENUM_TYPES[table_name] || {},
+    :entity_modified_columns => ENTITY_MODIFIED_COLUMNS[:table_name] || {}
+}.freeze
+
+def to_swagger_name(snake_case, is_class_name)
+  result =  snake_case.split('_').collect(&:capitalize).join
+  unless is_class_name
+    result[0] = result[0].downcase
+  end
+  result
+end
 
-columns = describe_rows.filter{ |row| include_field?(excluded_fields, row[0])} \
-  .map{ |row| {:name => row[0], :mysql_type => row[1], :big_query_type => to_bq_type(row[1])} }
+def to_property_name(column_name)
+  to_swagger_name(column_name, false)
+end
+
+COLUMNS = describe_rows.filter{ |row| include_field?(excluded_fields, row[0]) } \
+  .map{ |row| \
+    col_name = row[0]
+    mysql_type = simple_mysql_type(row[1])
+    type_override = TABLE_INFO[:enum_column_info][col_name]
+    type_info = MYSQL_TO_TYPES[mysql_type]
+    {
+        :name => col_name, \
+        :lambda_var => table_name[0].downcase, \
+        :mysql_type => mysql_type, \
+        :big_query_type => type_override ? type_override[:bigquery] : type_info[:bigquery], \
+        :is_enum => !type_override.nil?,
+        :java_type => type_override ? type_override[:java] : type_info[:java], \
+        :java_field_name => "#{to_camel_case(col_name, false)}", \
+        :java_constant_name => "#{table_name.upcase}__#{col_name.upcase}", \
```

Comment:
The double underscores between the table and column make it easier to read if the table name has a single underscore.

Reply:
This comment and the above comment should be in code rather than on the PR